latte/models/rlace/estimation.py

```diff
@@ -60,13 +65,25 @@ def fit(X: np.ndarray, Z: np.ndarray, rlace_params: Dict[str, Any], p_train: flo
     X_test_projected_A, X_test_projected_E = X_test @ A_hat_x, X_test @ E_hat_x
 
     # Construct the result
+    rng = np.random.default_rng(rng)
+    mi_estimation_ix = rng.choice(len(Z_test), min(8000, len(Z_test)), replace=False)
     mi_estimate = RLACEResult(
-        mutual_information=naive_ksg.mutual_information([X_test_projected_A.numpy(), Z_test.reshape(-1, 1)], k=3),
-        complement_mutual_information=naive_ksg.mutual_information(
-            [X_test_projected_E.numpy(), Z_test.reshape(-1, 1)], k=3
-        ),
+        mutual_information=ksg.estimate_mi_ksg(
+            X_test_projected_A[mi_estimation_ix].numpy(),
+            Z_test[mi_estimation_ix].reshape(-1, 1).numpy()
+            if isinstance(Z_test, torch.Tensor)
+            else Z_test[mi_estimation_ix].reshape(-1, 1),
+            neighborhoods=(5,),
```

Comment:
I'd refrain from hard-coded `5`, especially that now we have [an implementation of nice interface](https://github.com/pawel-czyz/bmi/blob/main/src/bmi/estimators/ksg.py). Could you please add a TODO comment with my name so I don't forget to change this?

Reply:
Will do!